### ISSUE TITLE
Why the loss function (mse) calculated by keras not the same as mine

### ISSUE THREAD

**Issue by yangzehao** [keras-team/tf-keras#651]
I want to test the loss function, mse in keras by myself. However, the calculated answers are different. The definition of mse is below: https://en.wikipedia.org/wiki/Mean_squared_error

The test code is below:
```python
from keras.datasets import boston_housing
import numpy as np
(train_data, train_targets), (test_data, test_targets) = boston_housing.load_data()


x_train = train_data.astype(np.float32)

from keras import models 
from keras import layers

model = models.Sequential() 
model.add(layers.Dense(64, activation='relu', input_shape=(13,))) 
model.add(layers.Dense(64, activation='relu')) 
model.add(layers.Dense(1))
model.compile(optimizer='rmsprop',loss='mse', metrics=['mae'])

y_train = train_targets.astype(np.float32)
# y_test = test_targets.astype(np.float32)

model.fit(x_train,y_train,epochs=1,batch_size=404)

print(np.mean((y_train - model.predict(x_train).ravel()) ** 2))
```
It shows that the loss function is around 816 in keras. However, from the definition of mse, the results is around 704. Why are the results different here?

**Comment by tilakrayal** [comment]
@lucasdavid,
The loss function mse has been reduced to 525 in Keras with the latest tensorflow and the keras. Kindly find the gist of it [here](https://colab.research.google.com/gist/tilakrayal/16b2eb42eecf5de924733d881eff06f2/untitled2370.ipynb).
Thank you!
